<!DOCTYPE html>

<html>

<head>

<meta charset="utf-8" />
<meta name="generator" content="pandoc" />
<meta http-equiv="X-UA-Compatible" content="IE=EDGE" />


<meta name="author" content="bvigil" />


<title>GBS</title>

<script src="site_libs/header-attrs-2.11/header-attrs.js"></script>
<script src="site_libs/jquery-3.6.0/jquery-3.6.0.min.js"></script>
<meta name="viewport" content="width=device-width, initial-scale=1" />
<link href="site_libs/bootstrap-3.3.5/css/journal.min.css" rel="stylesheet" />
<script src="site_libs/bootstrap-3.3.5/js/bootstrap.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/html5shiv.min.js"></script>
<script src="site_libs/bootstrap-3.3.5/shim/respond.min.js"></script>
<style>h1 {font-size: 34px;}
       h1.title {font-size: 38px;}
       h2 {font-size: 30px;}
       h3 {font-size: 24px;}
       h4 {font-size: 18px;}
       h5 {font-size: 16px;}
       h6 {font-size: 12px;}
       code {color: inherit; background-color: rgba(0, 0, 0, 0.04);}
       pre:not([class]) { background-color: white }</style>
<script src="site_libs/navigation-1.1/tabsets.js"></script>
<link href="site_libs/highlightjs-9.12.0/textmate.css" rel="stylesheet" />
<script src="site_libs/highlightjs-9.12.0/highlight.js"></script>

<style type="text/css">
  code{white-space: pre-wrap;}
  span.smallcaps{font-variant: small-caps;}
  span.underline{text-decoration: underline;}
  div.column{display: inline-block; vertical-align: top; width: 50%;}
  div.hanging-indent{margin-left: 1.5em; text-indent: -1.5em;}
  ul.task-list{list-style: none;}
    </style>

<style type="text/css">code{white-space: pre;}</style>
<script type="text/javascript">
if (window.hljs) {
  hljs.configure({languages: []});
  hljs.initHighlightingOnLoad();
  if (document.readyState && document.readyState === "complete") {
    window.setTimeout(function() { hljs.initHighlighting(); }, 0);
  }
}
</script>








<style type = "text/css">
.main-container {
  max-width: 940px;
  margin-left: auto;
  margin-right: auto;
}
img {
  max-width:100%;
}
.tabbed-pane {
  padding-top: 12px;
}
.html-widget {
  margin-bottom: 20px;
}
button.code-folding-btn:focus {
  outline: none;
}
summary {
  display: list-item;
}
pre code {
  padding: 0;
}
</style>



<!-- tabsets -->

<style type="text/css">
.tabset-dropdown > .nav-tabs {
  display: inline-table;
  max-height: 500px;
  min-height: 44px;
  overflow-y: auto;
  border: 1px solid #ddd;
  border-radius: 4px;
}

.tabset-dropdown > .nav-tabs > li.active:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li.active:before {
  content: "&#xe258;";
  border: none;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open:before {
  content: "";
  font-family: 'Glyphicons Halflings';
  display: inline-block;
  padding: 10px;
  border-right: 1px solid #ddd;
}

.tabset-dropdown > .nav-tabs > li.active {
  display: block;
}

.tabset-dropdown > .nav-tabs > li > a,
.tabset-dropdown > .nav-tabs > li > a:focus,
.tabset-dropdown > .nav-tabs > li > a:hover {
  border: none;
  display: inline-block;
  border-radius: 4px;
  background-color: transparent;
}

.tabset-dropdown > .nav-tabs.nav-tabs-open > li {
  display: block;
  float: none;
}

.tabset-dropdown > .nav-tabs > li {
  display: none;
}
</style>

<!-- code folding -->




</head>

<body>


<div class="container-fluid main-container">




<div id="header">



<h1 class="title toc-ignore">GBS</h1>
<h4 class="author">bvigil</h4>
<h4 class="date">7/12/2021</h4>

</div>


<div id="aplicaciones" class="section level1">
<h1><a href="#" class="btn btn-primary btn-lg"><strong>APLICACIONES</strong></a></h1>
<div>
<p style="text-align:center;">
<img src="Applications-of-Genotyping-by-Sequencing-integrating-broad-areas-of-genetics-and.png" width="600px">
</p>
</div>
</div>
<div id="pipeline-gbs-procesing" class="section level1">
<h1><a href="#" class="btn btn-primary btn-lg"><strong>PIPELINE GBS-PROCESING</strong></a></h1>
<div>
<p style="text-align:center;">
<img src="pipeline_GBS.png" width="1000px">
</p>
</div>
</div>
<div id="stacks" class="section level1">
<h1><a href="#" class="btn btn-primary btn-lg"><strong>STACKS</strong></a></h1>
<p><a href="https://catchenlab.life.illinois.edu/stacks/">Descargar</a></p>
<p><a href="https://catchenlab.life.illinois.edu/stacks/manual/#install">Instalar-manual</a></p>
<div>
<p style="text-align:center;">
<img src="stacks_full_pipelines.png" width="700px">
</p>
</div>
</div>
<div id="demultiplexación---filtrado---stacks" class="section level1">
<h1><a href="#" class="btn btn-primary btn-lg"><strong>DEMULTIPLEXACIÓN - FILTRADO - STACKS</strong></a></h1>
<p><a href="https://catchenlab.life.illinois.edu/stacks/manual-v1/"><strong>process_radtags</strong></a></p>
<div id="rawdata" class="section level2">
<h2><strong>1.RawData</strong></h2>
<div>
<p style="text-align:center;">
<img src="rawdata.png" width="600px">
</p>
</div>
</div>
<div id="barcodes-para-cada-individuo" class="section level2">
<h2><strong>2.Barcodes para cada individuo</strong></h2>
<div>
<p style="text-align:center;">
<img src="barcodes.png" width="200px">
</p>
</div>
</div>
<div id="demultiplexado-o-individualización" class="section level2">
<h2><strong>3.Demultiplexado o individualización</strong></h2>
<div>
<p style="text-align:center;">
<img src="demultiplexado.png" width="400px">
</p>
</div>
</div>
</div>
<div id="alineamiento-a-genoma-de-referencia" class="section level1">
<h1><a href="#" class="btn btn-primary btn-lg"><strong>ALINEAMIENTO A GENOMA DE REFERENCIA</strong></a></h1>
<div id="instalación-de-programas" class="section level2">
<h2><strong>1.INSTALACIÓN DE PROGRAMAS</strong></h2>
<p>BWA - BORROWS WHEELER ALIGNER</p>
<pre><code>sudo apt-get update
sudo apt-get install bwa</code></pre>
<p>SAMTOOLS</p>
<pre><code>sudo apt-get update
sudo apt-get install samtools</code></pre>
</div>
<div id="indexado-del-genoma-de-referencia" class="section level2">
<h2><strong>2.INDEXADO DEL GENOMA DE REFERENCIA</strong></h2>
<pre><code>#!/bin/bash
src=&quot;/home/bvigil/GBS_SEMINARIO/genoma&quot;
bwa index -p manihot-index -a bwtsw $src/GCF_001659605.2_M.esculenta_v8_genomic.fna</code></pre>
</div>
<div id="alineamiento" class="section level2">
<h2><strong>3.ALINEAMIENTO</strong></h2>
<pre><code>#!/bin/bash
src=&quot;/home/bvigil/GBS_SEMINARIO/&quot;
bwa_db=&quot;/home/bvigil/GBS_SEMINARIO/genoma/manihot-index&quot;

files=&quot;ME_038_S9_R1_001
ME_094_S10_R1_001
ME_171_S11_R1_001
ME_272_S12_R1_001
ME_368_S13_R1_001
ME_590_S14_R1_001
ME_674_S15_R1_001
ME_681_S16_R1_001&quot;

for ME in $files
do
        bwa mem -t 12 $bwa_db $src/data/${ME}.fastq.gz &gt; $src/alineamiento/${ME}.sam
done
</code></pre>
</div>
<div id="sam-bam" class="section level2">
<h2><strong>4.SAM –&gt; BAM</strong></h2>
<pre><code>#!/bin/bash
src=&quot;/home/bvigil/GBS_SEMINARIO/&quot;

files=&quot;ME_038_S9_R1_001
ME_094_S10_R1_001
ME_171_S11_R1_001
ME_272_S12_R1_001
ME_368_S13_R1_001
ME_590_S14_R1_001
ME_674_S15_R1_001
ME_681_S16_R1_001&quot;

for ME in $files
do
        samtools view -b -S $src/alineamiento/${ME}.sam -o $src/alineamiento/${ME}.bam
done</code></pre>
</div>
<div id="sort" class="section level2">
<h2><strong>5.SORT</strong></h2>
<pre><code>#!/bin/bash
src=&quot;/home/bvigil/GBS_SEMINARIO/alineamiento/&quot;

files=&quot;ME_038_S9_R1_001
ME_094_S10_R1_001
ME_171_S11_R1_001
ME_272_S12_R1_001
ME_368_S13_R1_001
ME_590_S14_R1_001
ME_674_S15_R1_001
ME_681_S16_R1_001&quot;

for ME in $files
do
        samtools sort $src/${ME}.bam -o $src/sort/${ME}.bam
done</code></pre>
</div>
</div>
<div id="gstacks---stacks" class="section level1">
<h1><a href="#" class="btn btn-primary btn-lg"><strong>GSTACKS - STACKS</strong></a></h1>
<pre><code>#!/bin/bash

src=&quot;/home/bvigil/GBS_SEMINARIO&quot;
gstacks -I $src/alineamiento/sort/ -M $src/scripts/popmap-yuca-piloto -O $src/stacks -t 24</code></pre>
</div>
<div id="populations---vcf" class="section level1">
<h1><a href="#" class="btn btn-primary btn-lg"><strong>POPULATIONS - VCF</strong></a></h1>
<p><strong>SNP-CALLING</strong></p>
<pre><code>#!/bin/bash

src=&quot;/home/bvigil/GBS_SEMINARIO&quot;
populations -P $src/stacks/ -M $src/scripts/popmap-yuca-piloto --vcf --plink -t 24</code></pre>
</div>
<div id="formato" class="section level1">
<h1><a href="#" class="btn btn-primary btn-lg"><strong>FORMATO</strong></a></h1>
<p><a href="https://samtools.github.io/hts-specs/VCFv4.2.pdf">VCF</a></p>
<p>VCF es un formato de archivo de texto (lo más probable es que se almacene de forma comprimida). Contiene líneas de metainformación, una línea de encabezado y luego líneas de datos, cada una de las cuales contiene información sobre una posición en el genoma. El formato también tiene la capacidad de contener información de genotipo en muestras para cada posición.</p>
<div>
<p style="text-align:center;">
<img src="VCF_FORMAT.png" width="1000px">
</p>
</div>
</div>
<div id="programas--filtrado-de-snps" class="section level1">
<h1><a href="#" class="btn btn-primary btn-lg"><strong>PROGRAMAS -FILTRADO DE SNPs</strong></a></h1>
<div id="vcftools" class="section level2">
<h2><strong>1.VCFTOOLS</strong></h2>
<pre><code>sudo apt-get update
sudo apt-get install vcftools</code></pre>
</div>
<div id="bcftools" class="section level2">
<h2><strong>2.BCFTOOLS</strong></h2>
<pre><code>sudo apt-get update
sudo apt-get install bcftools</code></pre>
</div>
</div>
<div id="descargar-archivo---demo" class="section level1">
<h1><a href="#" class="btn btn-primary btn-lg"><strong>DESCARGAR ARCHIVO - DEMO</strong></a></h1>
<p>Descargar <a href="https://drive.google.com/drive/folders/18xm4UcUeKhOCNCGLsWXVWW3jCQtKgwoV?usp=sharing">archivo VCF</a></p>
<div>
<p style="text-align:center;">
<img src="vcf_yuca.png" width="1000px">
</p>
</div>
</div>
<div id="filtrado-a-nivel-de-loci-variante-snp" class="section level1">
<h1><a href="#" class="btn btn-primary btn-lg"><strong>FILTRADO A NIVEL DE LOCI-VARIANTE-SNP</strong></a></h1>
<p>Solo se mantendran a variantes que se hayan genotipado con éxito en el 50% de los individuos de nuestra población.</p>
<pre class="bash"><code>vcftools --vcf /home/bvigil/GBS_SEMINARIO/stacks/yuca.chr14.recode.vcf --max-missing 0.5 --recode --recode-INFO-all --out /home/bvigil/GBS_SEMINARIO/stacks/yuca.chr14.maxmissing0.5 </code></pre>
<pre><code>## 
## VCFtools - 0.1.16
## (C) Adam Auton and Anthony Marcketta 2009
## 
## Parameters as interpreted:
##  --vcf /home/bvigil/GBS_SEMINARIO/stacks/yuca.chr14.recode.vcf
##  --recode-INFO-all
##  --max-missing 0.5
##  --out /home/bvigil/GBS_SEMINARIO/stacks/yuca.chr14.maxmissing0.5
##  --recode
## 
## After filtering, kept 8 out of 8 Individuals
## Outputting VCF file...
## After filtering, kept 5749 out of a possible 8110 Sites
## Run Time = 0.00 seconds</code></pre>
</div>
<div id="filtrado-por-individuos" class="section level1">
<h1><a href="#" class="btn btn-primary btn-lg"><strong>FILTRADO POR INDIVIDUOS</strong></a></h1>
<p>Este paso es para eliminar a los individuos que no se secuenciaron bien, que presentan datos faltantes por loci. Esto se realiza evaluando los niveles individuales de datos faltantes.</p>
<pre class="bash"><code>vcftools --vcf /home/bvigil/GBS_SEMINARIO/stacks/yuca.chr14.maxmissing0.5.recode.vcf --missing-indv</code></pre>
<pre><code>## 
## VCFtools - 0.1.16
## (C) Adam Auton and Anthony Marcketta 2009
## 
## Parameters as interpreted:
##  --vcf /home/bvigil/GBS_SEMINARIO/stacks/yuca.chr14.maxmissing0.5.recode.vcf
##  --missing-indv
## 
## After filtering, kept 8 out of 8 Individuals
## Outputting Individual Missingness
## After filtering, kept 5749 out of a possible 5749 Sites
## Run Time = 0.00 seconds</code></pre>
<p>revisamos el out.imiss</p>
<div>
<p style="text-align:center;">
<img src="imiss.png" width="500px">
</p>
</div>
<p>Observamos la data faltante por individuo en un histograma</p>
<pre><code>#!/bin/bash

mawk &#39;!/IN/&#39; /home/bvigil/GBS_SEMINARIO/stacks/out.imiss | cut -f5 &gt; /home/bvigil/GBS_SEMINARIO/stacks/totalmissing
gnuplot &lt;&lt; \EOF 
set terminal dumb size 120, 30
set autoscale 
unset label
set title &quot;Histogram of % missing data per individual&quot;
set ylabel &quot;Number of Occurrences&quot;
set xlabel &quot;% of missing data&quot;
#set yr [0:100000]
binwidth=0.01
bin(x,width)=width*floor(x/width) + binwidth/2.0
plot &#39;totalmissing&#39; using (bin($1,binwidth)):(1.0) smooth freq with boxes
pause -1
EOF</code></pre>
<pre><code>./hist_miss.sh</code></pre>
<div>
<p style="text-align:center;">
<img src="totalmissing.png" width="700px">
</p>
</div>
</div>
<div id="filtrado-por-frecuencia-de-alelo-menor---maf" class="section level1">
<h1><a href="#" class="btn btn-primary btn-lg"><strong>FILTRADO POR FRECUENCIA DE ALELO MENOR - MAF</strong></a></h1>
<p>Este filtrado es importante y depende de los objetivos del estudio, el tipo y la cantidad de individuos de la población. Por lo general se hace un filtrado de maf 0.01, esto significa que se eliminaran aquellos loci cuyo alelo menor tenga una frecuencia menor al 1%.</p>
<pre class="bash"><code>vcftools --vcf /home/bvigil/GBS_SEMINARIO/stacks/yuca.chr14.maxmissing0.5.recode.vcf --maf 0.01 --recode --recode-INFO-all --out /home/bvigil/GBS_SEMINARIO/stacks/yuca.chr14.maxmissing0.5.maf0.01</code></pre>
<pre><code>## 
## VCFtools - 0.1.16
## (C) Adam Auton and Anthony Marcketta 2009
## 
## Parameters as interpreted:
##  --vcf /home/bvigil/GBS_SEMINARIO/stacks/yuca.chr14.maxmissing0.5.recode.vcf
##  --recode-INFO-all
##  --maf 0.01
##  --out /home/bvigil/GBS_SEMINARIO/stacks/yuca.chr14.maxmissing0.5.maf0.01
##  --recode
## 
## After filtering, kept 8 out of 8 Individuals
## Outputting VCF file...
## After filtering, kept 5749 out of a possible 5749 Sites
## Run Time = 0.00 seconds</code></pre>
</div>
<div id="analisis-de-estructura-poblacional" class="section level1">
<h1><a href="#" class="btn btn-primary btn-lg"><strong>ANALISIS DE ESTRUCTURA POBLACIONAL</strong></a></h1>
<div id="activar-paquetes-en-r" class="section level2">
<h2>1.- ACTIVAR PAQUETES EN R :</h2>
<pre class="r"><code>library(&quot;adegenet&quot;)</code></pre>
<pre><code>## Loading required package: ade4</code></pre>
<pre><code>## 
##    /// adegenet 2.1.5 is loaded ////////////
## 
##    &gt; overview: &#39;?adegenet&#39;
##    &gt; tutorials/doc/questions: &#39;adegenetWeb()&#39; 
##    &gt; bug reports/feature requests: adegenetIssues()</code></pre>
<pre class="r"><code>library(&quot;vcfR&quot;)</code></pre>
<pre><code>## 
##    *****       ***   vcfR   ***       *****
##    This is vcfR 1.12.0 
##      browseVignettes(&#39;vcfR&#39;) # Documentation
##      citation(&#39;vcfR&#39;) # Citation
##    *****       *****      *****       *****</code></pre>
<pre class="r"><code>library(&quot;ggplot2&quot;)</code></pre>
<pre class="r"><code>### cargar el archivo VCF filtrado####
yuca_vcf&lt;-read.vcfR(&quot;yuca.chr14.maxmissing0.5.maf0.01.recode.vcf&quot;)</code></pre>
<pre><code>## Scanning file to determine attributes.
## File attributes:
##   meta lines: 14
##   header_line: 15
##   variant count: 5749
##   column count: 17
## 
Meta line 14 read in.
## All meta lines processed.
## gt matrix initialized.
## Character matrix gt created.
##   Character matrix gt rows: 5749
##   Character matrix gt cols: 17
##   skip: 0
##   nrows: 5749
##   row_num: 0
## 
Processed variant 1000
Processed variant 2000
Processed variant 3000
Processed variant 4000
Processed variant 5000
Processed variant: 5749
## All variants processed</code></pre>
<pre class="r"><code>####convertir a formato genind ####
gl_yuca&lt;-vcfR2genlight(yuca_vcf)
###indicar ploidia####
ploidy(gl_yuca)&lt;-2
###cargar el mapa de poblacion ###
mapa_pop&lt;-read.csv(&quot;mapa2.csv&quot;, sep = &quot;,&quot;, header = T)
pop(gl_yuca)&lt;-mapa_pop$provincia
####añadimos estratificación ####
strata(gl_yuca)&lt;-data.frame(mapa_pop$provincia)
nameStrata(gl_yuca)&lt;-~provincia
table(strata(gl_yuca, ~provincia))</code></pre>
<pre><code>## 
##    huanuco       lima     tumbes   arequipa    ucayali      cusco      junin 
##          1          1          1          1          1          1          1 
## san-martin 
##          1</code></pre>
<pre class="r"><code>#### Análisis multivariado PCA####
clust_yuca&lt;-tab(gl_yuca, freq=T, NA.method=&quot;mean&quot;)
pca_yuca&lt;-dudi.pca(clust_yuca, scale = T, scannf = F)

library(&quot;RColorBrewer&quot;)
cols &lt;- brewer.pal(n = nPop(gl_yuca), name = &quot;Dark2&quot;)

#### graficar PCA####
library(&quot;factoextra&quot;)</code></pre>
<pre><code>## Welcome! Want to learn more? See two factoextra-related books at https://goo.gl/ve3WBa</code></pre>
<pre class="r"><code>fviz_pca_ind(pca_yuca, geom.ind = &quot;point&quot;, col.ind = gl_yuca$pop, palette = cols, addEllipses = FALSE, ellipse.type=&quot;confidence&quot;, legend.title = &quot;Provincias&quot;, pointshape=1, cpoint=1, xlab=&quot;PC1&quot;, ylab=&quot;PC2&quot;)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-7-1.png" width="672" /></p>
<pre class="r"><code>fviz_pca_ind(pca_yuca, col.ind = &quot;cos2&quot;, gradient.cols = c(&quot;#00AFBB&quot;, &quot;#E7B800&quot;, &quot;#FC4E07&quot;),repel = FALSE, label=&quot;none&quot;, addEllipses = FALSE, ellipse.type=&quot;euclid&quot;, xlab=&quot;PC1&quot;, ylab=&quot;PC2&quot;, pointshape=19, cpoint=1)</code></pre>
<p><img src="index_files/figure-html/unnamed-chunk-8-1.png" width="672" /></p>
</div>
</div>




</div>

<script>

// add bootstrap table styles to pandoc tables
function bootstrapStylePandocTables() {
  $('tr.odd').parent('tbody').parent('table').addClass('table table-condensed');
}
$(document).ready(function () {
  bootstrapStylePandocTables();
});


</script>

<!-- tabsets -->

<script>
$(document).ready(function () {
  window.buildTabsets("TOC");
});

$(document).ready(function () {
  $('.tabset-dropdown > .nav-tabs > li').click(function () {
    $(this).parent().toggleClass('nav-tabs-open');
  });
});
</script>

<!-- code folding -->


<!-- dynamically load mathjax for compatibility with self-contained -->
<script>
  (function () {
    var script = document.createElement("script");
    script.type = "text/javascript";
    script.src  = "https://mathjax.rstudio.com/latest/MathJax.js?config=TeX-AMS-MML_HTMLorMML";
    document.getElementsByTagName("head")[0].appendChild(script);
  })();
</script>

</body>
</html>
